client/my-sites/site-settings/form-general.jsx

```diff
@@ -286,71 +357,73 @@ const FormGeneral = React.createClass( {
 						<FormRadio
 							name="blog_public"
 							value="-1"
-							checked={ -1 === parseInt( this.state.blog_public, 10 ) }
+							checked={ -1 === parseInt( fields.blog_public, 10 ) }
 							onChange={ this.handleRadio }
-							disabled={ this.state.fetchingSettings }
+							disabled={ isRequestingSettings }
 							onClick={ this.onRecordEvent( 'Clicked Site Visibility Radio Button' ) } />
-						<span>{ this.translate( 'Private' ) }</span>
+						<span>{ translate( 'Private' ) }</span>
 						<FormSettingExplanation isIndented>
-							{ this.translate( 'Your site is only visible to you and users you approve.' ) }
+							{ translate( 'Your site is only visible to you and users you approve.' ) }
 						</FormSettingExplanation>
 					</FormLabel>
 				}
 
 			</FormFieldset>
 		);
-	},
+	}
 
-	handleAmpToggle() {
-		this.setState( { amp_is_enabled: ! this.state.amp_is_enabled }, () => {
+	handleAmpToggle = () => {
+		const { fields, updateFields } = this.props;
+		updateFields( { amp_is_enabled: ! fields.amp_is_enabled }, () => {
 			this.submitForm();
-			this.onRecordEvent( 'Clicked AMP Toggle' );
+			this.onRecordEvent( 'Clicked AMP Toggle' )();
```

Comment:
I'll agree here and put in a nod to some nomenclature I tend to favor. `mapDispatchToProps` is an excellent place to map between the analytics API and the component API. I always ask myself what the component itself wants to do or cares about. does it care about tracking general analytics data? probably not. in this case we might be on the extreme end of actually caring about tracking things, but ideally I think we would be using the utilities like `withAnalytics` to wrap _actual actions_ that we care about with the incidental analytics calls.

in this case, maybe we could remove the use of `onRecordEvent` and inject into the props a function called something like `trackClick( clickTargetName )` and then we would see `this.props.trackClick( 'AMP Toggle' )`.

otherwise, if something like `submitForm` implied to us at an app level that we should be tracking it, maybe the analytics can be pulled outside and wrapped about the form submitting code.